File: onnxruntime/examples/issue22.py
# brew install libomp
# python -m venv .venv
# source .venv/bin/activate
# python ./issue22.py

import onnxruntime # v1.2.0

session = onnxruntime.InferenceSession("model.onnx")

print("Inputs:")
for idx, inputs in enumerate(session.get_inputs()):
    print("idx:", idx)
    print("   Name:", inputs.name)
    print("   Shape:", inputs.shape)
print("Outputs:")
for idx, outputs in enumerate(session.get_outputs()):
    print("idx:", idx)
    print("   Name:", outputs.name)
    print("   Shape:", outputs.shape)

outputs = session.run(None, {"input_ids": [[1, 2, 3]], "attention_mask": [[1, 1, 1]]})[0]
print(outputs.shape)

File: onnxruntime/examples/issue22.rs
//! Example reproducing issue #22.
//!
//! `model.onnx` available to download here:
//! https://drive.google.com/file/d/1FmL-Wpm06V-8wgRqvV3Skey_X98Ue4D_/view?usp=sharing

use ndarray::Array2;
use onnxruntime::{environment::Environment, tensor::OrtOwnedTensor, GraphOptimizationLevel};

fn main() {
    let env = Environment::builder().with_name("env").build().unwrap();
    let mut session = env
        .new_session_builder()
        .unwrap()
        .with_optimization_level(GraphOptimizationLevel::Basic)
        .unwrap()
        .with_model_from_file("model.onnx")
        .unwrap();

    println!("{:#?}", session.inputs);
    println!("{:#?}", session.outputs);

    let input_ids = Array2::<f32>::from_shape_vec((1, 3), vec![1f32, 2f32, 3f32]).unwrap();
    let attention_mask = Array2::<f32>::from_shape_vec((1, 3), vec![1f32, 1f32, 1f32]).unwrap();

    let outputs: Vec<OrtOwnedTensor<f32, _>> =
        session.run(vec![input_ids, attention_mask]).unwrap();
    print!("outputs: {:#?}", outputs);
}
